llvm/lib/CodeGen/LiveVariables.cpp

```diff
@@ -534,9 +534,7 @@ void LiveVariables::runOnInstr(MachineInstr &MI,
         UseRegs.push_back(MOReg);
     } else {
       assert(MO.isDef());
-      // FIXME: We should not remove any dead flags. However the MIPS RDDSP
-      // instruction needs it at the moment: http://llvm.org/PR27116.
-      if (MOReg.isPhysical() && !MRI->isReserved(MOReg))
+      if (!(MOReg.isPhysical() && MRI->isReserved(MOReg)))
```

Comment:
This isn't removing the hack, this only removes the comment and regresses the style of the condition

Reply:
It's not the same condition.

Reply:
The condition should be demorganed. The comment also states that dead flags should not be removed, and this is still removing a dead flag

Reply:
It just matches the condition in the branch above it

And I actually checked the history of this file, and this was like that before the change